module/x/gravity/keeper/keeper_delegate_key.go

```diff
@@ -109,6 +109,23 @@ func (k Keeper) GetOrchestratorValidator(ctx sdk.Context, orch sdk.AccAddress) (
 	return validator, true
 }
 
+// GetOrchestratorValidatorAddr returns the validator address associated with an orchestrator key.
+// Getting a result from this function means that the validator existed at some point and sent a SetOrchestratorAddress
+// message. It does not mean that the validator is in the current validator set, for that use GetOrchestratorValidator.
+// This will hold true as long as we never delete any delegate keys.
+func (k Keeper) GetOrchestratorValidatorAddr(ctx sdk.Context, orch sdk.AccAddress) (validator sdk.ValAddress, found bool) {
```

Comment:
This is a nice function, it should also be used above this in GetOrchestratorValidator